test/test_other.py

```diff
@@ -8633,25 +8633,34 @@ def test(infile, source_map_added_dir=''):
   def test_emsymbolizer(self):
     # Test DWARF output
     self.run_process([EMCC, test_file('core/test_dwarf.c'),
-                      '-g', '-O1', '-o', 'test_dwarf.js'])
+                      '-g', '-gsource-map', '-O1', '-o', 'test_dwarf.js'])
```

Comment:
It might be a good idea to also test these in separate invocations, just to be sure there is no interaction. (since using both at once is pretty rare, it seems safer to also test closer to what most users do)

Reply:
Yeah I can do that but the addresses have to change.

Reply:
> After it passes `wasm-emscripten-finalize`, the starting offsets to sections change, causing the address of the instructions change too

After reading this (and after I had to adjust the offsets in #17527) it occurred to me that maybe we should just do most of the emsymbolizer tests using the `--addrtype=code` flag, to make it use section offsets instead of file offsets. That will make it easier to compare, and make it more robust to unrelated changes. We should still have at least one tests that covers the file-offset mode, but it can be separate.